server/src/main/java/org/opensearch/repositories/RepositoriesService.java

```diff
@@ -627,6 +628,14 @@ public static void validateRepositoryMetadataSettings(
                     + minVersionInCluster
             );
         }
+        if (REMOTE_STORE_INDEX_SHALLOW_COPY.get(repositoryMetadataSettings) && FeatureFlags.isEnabled(FeatureFlags.REMOTE_STORE)) {
```

Comment:
the second condition should be `!FeatureFlags.isEnabled(FeatureFlags.REMOTE_STORE)` **not** is missing